src/autoTune0.ml

```diff
@@ -91,6 +91,7 @@ let collectFactors visitAction visitedObject =
   ignore (visitAction visitor visitedObject);
   factors
 
-let is_large_array = function
-  | TArray (_,Some (Const (CInt (i,_,_))),_) -> i > Z.of_int @@ 10 * get_int "ana.base.arrays.unrolling-factor"
+let is_large_array t =
+  match Cil.unrollType t with
+  | TArray (_,Some (Const (CInt (i,_,_))),_) -> i > Z.of_int @@ 10 * get_int "ana.base.arrays.unrolling-factor" (* TODO: Cil.lenOfArray but with Z.t? *)
```

Comment:
Or use `Cil.lenofArray` and do it on `int`? Arrays of length greater than $2^63$ are not common anyway.

Reply:
I think I considered but it didn't seem much nicer. I changed it nevertheless because `Cil.lenOfArray` does constant folding which could be a good idea (for array lengths written as constant expressions via `#define`-s).
I'll leave the TODO though because it's silly on the CIL side: the computation is done with `Z.t` and downcast...